packages/core/src/ckb/transaction.ts

```diff
@@ -1595,6 +1595,40 @@ export class Transaction extends mol.Entity.Base<
     this.setWitnessArgsAt(position, witness);
   }
 
+  /**
+   * Prepare multisig witness
+   *
+   * @param scriptLike - The script associated with the transaction, represented as a ScriptLike object.
+   * @param multisigMetadata - The metadata of the multisig script.
+   * @param threshold - The threshold of the multisig script.
+   * @param client - The client for complete extra infos in the transaction.
+   * @returns A promise that resolves to the prepared transaction
+   *
+   * @example
+   * ```typescript
+   * await tx.prepareMultisigWitness(script, multisigMetadata, threshold, client);
+   * ```
+   */
+  async prepareMultisigWitness(
```

Comment:
It appears that this logic is closely coupled to the Multisig Signer. Maybe moving it to the Signer implementation is better than making it in the `Transaction`